include/net/nrf_cloud_rest.h

```diff
@@ -0,0 +1,266 @@
+
+	/** Results from API call */
+	/** HTTP status of API call */
+	enum nrf_cloud_http_status status;
+	/** Start of response data in rx_buf */
+	char *response;
+	/** Length of response data */
+	size_t response_len;
+};
+
+/** @brief Data required for nRF Cloud location request */
+struct nrf_cloud_rest_location_request {
+	/** Unique device identifier; null-terminated */
+	char *device_id;
+	/** Network information used in location request */
+	struct lte_lc_cells_info *net_info;
+};
+
+/** @brief Data required for nRF Cloud Assisted GPS (A-GPS) request */
+struct nrf_cloud_rest_agps_request {
+	/** Unique device identifier; null-terminated */
+	char *device_id;
+	enum nrf_cloud_rest_agps_req_type type;
+	/** Required for custom request type */
+	struct gps_agps_request *agps_req;
+	/** Optional; provide network info or set to NULL. The cloud cannot
+	 * provide location assistance data if network info is NULL.
+	 */
+	struct lte_lc_cells_info *net_info;
+};
+
+/** @brief nRF Cloud Assisted GPS (A-GPS) result */
+struct nrf_cloud_rest_agps_result {
+	/** User-provided buffer to hold AGPS data */
+	char *buf;
+	/** Size of user-provided buffer */
+	size_t buf_sz;
+	/** Size of the AGPS data copied into buffer */
+	size_t agps_sz;
+};
+
+/** @defgroup nrf_cloud_rest_pgps_omit Omit item from P-GPS request.
+ * @{
+ */
+#define NRF_CLOUD_REST_PGPS_REQ_NO_COUNT	0
+#define NRF_CLOUD_REST_PGPS_REQ_NO_INTERVAL	0
+#define NRF_CLOUD_REST_PGPS_REQ_NO_GPS_DAY	0
+#define NRF_CLOUD_REST_PGPS_REQ_NO_GPS_TOD	(-1)
+/** @} */
+
+/** @brief Data required for nRF Cloud Predicted GPS (P-GPS) request */
+struct nrf_cloud_rest_pgps_request {
+	/** Unique device identifier; null-terminated */
+	char *device_id;
+	/** Data to be included in the P-GPS request. To omit an item
+	 * use the appropriate define in @ref nrf_cloud_rest_pgps_omit
+	 */
+	const struct gps_pgps_request *pgps_req;
+};
+
+/**
+ * @brief nRF Cloud location request.
+ *
+ * @param[in,out] rest_ctx Context for communicating with nRF Cloud's REST API.
+ * @param[in]     request Data to be provided in API call.
+ * @param[in,out] result Optional; parsed results of API response.
+ *                       If NULL, user should inspect the response buffer of
+ *                       @ref nrf_cloud_rest_context.
+ *
+ * @retval 0 If successful.
+ *          Otherwise, a (negative) error code is returned.
+ */
+int nrf_cloud_rest_location_get(struct nrf_cloud_rest_context *const rest_ctx,
+	struct nrf_cloud_rest_location_request const *const request,
+	struct nrf_cloud_cell_pos_result *const result);
+
+/**
+ * @brief nRF Cloud Assisted GPS (A-GPS) data request.
+ *
+ * @param[in,out] rest_ctx Context for communicating with nRF Cloud's REST API.
+ * @param[in]     request Data to be provided in API call.
+ * @param[in,out] result Optional; Additional buffer for A-GPS data. This is
+ *			 necessary when the A-GPS data from the cloud is larger
+ *			 than @ref CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE.
```

Comment:
Apparently, you cannot use @ref here as it makes the build fail. See https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_styleguide.html#references

Reply:
is this a recent change to the doc builds?  using @ref this way only recently started to fail.  i see it used in other NCS source files.

what is the correct method?  
i will try:
```
:option:`CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE`
````

Reply:
:option: would work in an .rst doc, but most probably not in doxygen.

Reply:
Maybe this https://github.com/nrfconnect/sdk-nrf/pull/5124 is the reason for the surprises.